Tests/UnitTests/Core/Surfaces/LineSurfaceTests.cpp

```diff
@@ -266,6 +275,59 @@ BOOST_AUTO_TEST_CASE(LineSurfaceTransformRoundTripEtaStability) {
   }
 }
 
+BOOST_AUTO_TEST_CASE(LineSurfaceIntersection) {
+  using namespace Acts::UnitLiterals;
+
+  auto eps = 1e-10;
+
+  auto direction = Vector3(1, 1, 100).normalized();
+  auto boundVector =
+      (BoundVector() << 1_cm, 1_cm, VectorHelpers::phi(direction),
+       VectorHelpers::theta(direction), 1, 0)
+          .finished();
+  auto pathLimit = 1_cm;
+
+  auto surface = std::make_shared<LineSurfaceStub>(Transform3::Identity());
+
+  auto initialParams = BoundTrackParameters(surface, boundVector, 1);
+
+  auto propagator = Propagator<StraightLineStepper>({});
+
+  auto displacedParameters =
+      CurvilinearTrackParameters(Vector4::Zero(), Vector3::Zero(), 1);
+  {
+    auto options = PropagatorOptions<>(tgContext, {});
+    options.direction = Acts::Direction::Backward;
+    options.pathLimit = pathLimit;
+
+    auto result = propagator.propagate(initialParams, options);
+    BOOST_CHECK(result.ok());
+    BOOST_CHECK(result.value().endParameters);
+
+    displacedParameters = result.value().endParameters.value();
+  }
+
+  auto intersection =
+      surface->intersect(tgContext, displacedParameters.position(tgContext),
+                         displacedParameters.unitDirection());
+  CHECK_CLOSE_ABS(intersection.intersection.pathLength, pathLimit, eps);
+
+  auto endParameters = BoundTrackParameters(surface, BoundVector::Zero(), 1);
+  {
+    auto options = PropagatorOptions<>(tgContext, {});
+    options.direction = Acts::Direction::Forward;
+    options.maxStepSize = 1_mm;
+
+    auto result = propagator.propagate(displacedParameters, *surface, options);
+    BOOST_CHECK(result.ok());
+    BOOST_CHECK(result.value().endParameters);
+    CHECK_CLOSE_ABS(result.value().pathLength, pathLimit, eps);
+    endParameters = result.value().endParameters.value();
+  }
+
+  CHECK_CLOSE_ABS(initialParams.parameters(), endParameters.parameters(), eps);
```

Comment:
This I think makes more sense to check, i.e. that the propagation correctly uses the new intersect method to approach the surface in `1mm` steps.